File: src/Core/Spawning/Player.zig
const sustenet = @import("root").sustenet;
const Vector3 = sustenet.zig_numerics.vector.Vec3(f32);

const Spawnee = sustenet.core.spawning.Spawnee;
const Player = @This();

super: Spawnee,
username: []u8,

pub fn new(id: u32, username: []const u8, spawn_position: Vector3) Player {
    const player = Player{
        .super = Spawnee.new(id, spawn_position),
        .username = username,
    };
    return player;
}

File: rust/src/core/spawning/player.rs
use quaternion::Quaternion;
use vector3::Vector3;

use super::Spawnee;

pub struct Player {
    base: Spawnee,
    username: Vec<u8>,
}

impl Player {
    pub fn new(id: u32, username: &[u8], spawn_pos: Vector3, spawn_rot: Quaternion<f64>) -> Self {
        let base = Spawnee::new(id, spawn_pos, spawn_rot);
        Player {
            base,
            username: username.to_vec(),
        }
    }
}
